doc/cloud-init.md

```diff
@@ -31,6 +31,111 @@ For LXD instances, `vendor-data` should be used in profiles rather than the inst
 
 Cloud-config examples can be found here: https://cloudinit.readthedocs.io/en/latest/topics/examples.html
 
+# Working with cloud-init
+
+For a safe way to test, use a new profile that's copied from the default profile.
+
+```
+$ lxc profile copy default test
+```
+
+Then edit the new `test` profile. You may wish to set your `EDITOR` environment variable first.
+
+```
+$ lxc profile edit test
+```
+
+For a new LXD installation, the configuration file should be similar to
+```yaml
+config: {}
+description: Default LXD profile
+devices:
+  eth0:
+    name: eth0
+    network: lxdbr0
+    type: nic
+  root:
+    path: /
+    pool: default
+    type: disk
+```
+
+Once you've set up the `cloud-init` configuration, use `lxc launch` with `--profile <profilename>` to apply the profile to the instance.
+
+## Adding cloud-init keys to the configuration
+
+cloud-init keys require a specific syntax. You use a pipe symbol (`|`) to indicate that LXD should pipe the contents of the relevant key
+to cloud-init.
+```yaml
+config:
+  cloud-init.user-data: |
+```
+
+```yaml
+config:
+  cloud-init.vendor-data: |
+```
+
+```yaml
+config:
+  cloud-init.network-config: |
+```
+
+## Custom user-data configuration
+
+cloud-init uses the `user-data` (and `vendor-data`) section to do things like upgrade packages, install packages, or run arbitrary commands.
+
+A `cloud-init.user-data` key must have a first line of `#cloud-config`; without this marker, `cloud-init` will not process the remaining configuration.
```

Comment:
That's incorrect. Cloud-init supports multiple syntaxes, one of which happens to be `#cloud-config`

Reply:
See https://cloudinit.readthedocs.io/en/latest/topics/format.html for all the valid formats.

Reply:
Thanks. Section updated, linking to the formats.